src/util/simplify_expr_int.cpp

```diff
@@ -237,7 +240,10 @@ bool simplify_exprt::simplify_mult(exprt &expr)
 
   if(c_sizeof_type.is_not_nil())
   {
-    assert(found);
+    INVARIANT(
+      constant_found,
+      "c_sizeof_type is only set to a non-nil value "
+      "if a constant has been found");
```

Comment:
Hmm, having two entangled variables is questionable - I know, not your design choice, but still worth looking into I'd say.

Reply:
The redesign of this section to disentangle these two variables might be considered not in the scope of the PR.